File: interview_prep/algorithm/java/src/test/java/hoa/can/code/AddBinaryTest.java
package hoa.can.code;

import hoa.can.code.ez.AddBinary;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class AddBinaryTest {
    AddBinary sol = new AddBinary();

    @Test
    @DisplayName("11 + 1 = 100")
    public void test1() {
        assertEquals("100", sol.addBinary("11","1"));
    }

    @Test
    @DisplayName("1 + 11 = 100")
    public void test2() {
        assertEquals("100", sol.addBinary("1","11"));
    }

    @Test
    @DisplayName("11 + 1 = 100")
    public void test3() {
        assertEquals("10101", sol.addBinary("1010","1011"));
    }
}

File: interview_prep/algorithm/Kotlin/src/test/kotlin/AddBinaryTest.kt
import hoa.can.code.AddBinary
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import kotlin.test.DefaultAsserter.assertEquals

internal class AddBinaryTest {
    private val tc: AddBinary = AddBinary()

    @Test
    fun testSum() {
        assertEquals("11+1=100", tc.addBinary("11","1"),"100")
        Assertions.assertEquals("10101", tc.addBinary("1010", "1011"),"10101")
    }
}